src/sentry/search/events/base.py

```diff
@@ -39,5 +39,16 @@ def project_slugs(self) -> Mapping[str, int]:
 
         return {p.slug: p.id for p in project_slugs}
 
-    def column(self, name: str) -> Column:
-        return Column(self.resolve_column_name(name))
+    def column(self, name: str, alias: Optional[str] = None) -> Column:
```

Comment:
Thoughts on instead introducing a new `class AliasedColumn` to match what snql will eventually add?
since AliasedColumns won't be usable in Functions, we can preemptively handle that as we continue building with it?

Reply:
Ahh yes, just did this to get things working for now, will introduce a `AliasedColumn` soon 👍

Reply:
Instead of a class, i opted for just a function `aliased_column` that returns a `SelectType`. A class would've been just more boilerplate

Reply:
Actually could we update this so it doesn't accept the alias? Reason being that Function's likely won't accept aliased_columns as parameters and we'll want the plain resolved column in those cases.
(also add something along those lines to the docstring)

Reply:
Not sure what you mean by this. If it doesn't accept `alias`, how would it know that the column needs to be aliased? The `alias` arg is intentionally optional to handle the use case with functions where it doesn't need and may not work with the alias.

Reply:
resolved in 247dbfc